src/main/java/com/google/step/snippet/data/Auth.java

```diff
@@ -0,0 +1,25 @@
+package com.google.step.snippet.data;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+public class Auth {
```

Comment:
This class feels like a bit of needless indirection.

If you didn't use this Auth class, you would keep a UserService around. (Note that you don't even need to - UserServiceFactory.getUserService() is using the Singleton design pattern, meaning that after the first call to it, subsequent calls to it are essentially free, so you can call it as much as you want without performance loss.)

If you do use this Auth class, you need to keep an instance of it around. But all the methods are just wrappers, so I don't see any benefit.

Reply:
I edited the class so that it's no longer just wrappers. This reduces the need to the if checks for login within the servlet itself since the Auth class will take care of it and return the information that the servlet is _actually_ looking for since usually, the servlet will want to *do something* if the user is authenticated, not just checking _if_ the user is authenticated. I think this class will be useful for future features that will need to check if the user is logged in to do something.

Reply:
I can see the purpose of a shared class to handle retrieving the button url and text. But I am unconvinced there will be many other things to share. For example, only the feedback servlet will care about checking the user is authenticated prior to incrementing a vote tally.

I am also unconvinced of the statement that "the servlet will want to do something if the user is authenticated, not just checking if the user is authenticated". For example, the feedback servlet will want to do nothing if asked to upvote but the user is not authenticated (maybe it'll send back an error message). The account servlet will want to display an error page, but that's the job of the servlet, not a shared class.

Can you provide examples of what you're envisioning?

Reply:
Hmm I guess I had been thinking that the class would handle the error message for non authenticated user, but making the servlet class instead of the class do that makes sense. 

I this case, I can just keep the class for the button and text which will be shared across the Homepage, Search page, (and future accounts page)?